packages/rxfire/firestore/collection/index.ts

```diff
@@ -0,0 +1,165 @@
+
+const ALL_EVENTS: firestore.DocumentChangeType[] = [
+  'added',
+  'modified',
+  'removed'
+];
+
+/**
+ * Create an operator that determines if a the stream of document changes
+ * are specified by the event filter. If the document change type is not
+ * in specified events array, it will not be emitted.
+ */
+const filterEvents = (events?: firestore.DocumentChangeType[]) =>
+  filter((changes: firestore.DocumentChange[]) => {
+    let hasChange = false;
+    changes.forEach(change => {
+      if (events.indexOf(change.type) >= 0) {
+        hasChange = true;
+      }
+    });
+    return hasChange;
+  });
+
+/**
+ * Create an operator that filters out empty changes. We provide the
+ * ability to filter on events, which means all changes can be filtered out.
+ * This creates an empty array and would be incorrect to emit.
+ */
+const filterEmpty = filter(
+  (changes: firestore.DocumentChange[]) => changes.length > 0
+);
+
+/**
+ * Creates a new sorted array from a new change.
+ * @param combined
+ * @param change
+ */
+function combineChange<T>(
+  combined: firestore.DocumentChange[],
+  change: firestore.DocumentChange
+): firestore.DocumentChange[] {
+  switch (change.type) {
+    case 'added':
+      if (
+        combined[change.newIndex] &&
+        combined[change.newIndex].doc.id == change.doc.id
+      ) {
+        // Not sure why the duplicates are getting fired
+      } else {
+        combined.splice(change.newIndex, 0, change);
+      }
+      break;
+    case 'modified':
+      // When an item changes position we first remove it
+      // and then add it's new position
+      if (change.oldIndex !== change.newIndex) {
+        combined.splice(change.oldIndex, 1);
+        combined.splice(change.newIndex, 0, change);
+      } else {
+        combined.splice(change.newIndex, 1, change);
+      }
+      break;
+    case 'removed':
+      combined.splice(change.oldIndex, 1);
+      break;
+  }
+  return combined;
+}
+
+/**
+ * Combines the total result set from the current set of changes from an incoming set
+ * of changes.
+ * @param current
+ * @param changes
+ * @param events
+ */
+function combineChanges<T>(
+  current: firestore.DocumentChange[],
+  changes: firestore.DocumentChange[],
+  events: firestore.DocumentChangeType[] = ALL_EVENTS
+) {
+  changes.forEach(change => {
+    // skip unwanted change types
+    if (events.indexOf(change.type) > -1) {
+      current = combineChange(current, change);
+    }
+  });
+  return current;
+}
+
+/**
+ * Return a stream of document changes on a query. These results are not in sort order but in
+ * order of occurence.
+ * @param query
+ */
+export function docChanges(
+  query: firestore.Query,
+  events: firestore.DocumentChangeType[] = ALL_EVENTS
+) {
+  return fromCollectionRef(query).pipe(
+    map(snapshot => snapshot.docChanges()),
+    filterEvents(events),
+    filterEmpty
+  );
+}
+
+/**
+ * Return a stream of document changes on a query. These results are in sort order.
```

Comment:
This comment is inaccurate, we return a stream of `DocumentSnapshots` not changes.

Reply:
fixed.